bindings/go/generator/jsonschemagen/extractor_test.go

```diff
@@ -0,0 +1,86 @@
+package jsonschemagen
+
+import (
+	"go/ast"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestExtractStructDocPrefersTypeSpecDoc(t *testing.T) {
+	ts := &ast.TypeSpec{
+		Name: ast.NewIdent("Example"),
+		Doc: commentGroup(
+			"// +marker",
+			"// nolint:revive",
+			"// Primary description",
+			"//",
+			"// Deprecated: use another type",
+			"// trailing note",
+		),
+	}
+
+	gd := &ast.GenDecl{
+		Doc: commentGroup("// fallback description"),
+	}
+
+	desc, deprecated := extractStructDoc(ts, gd)
+
+	require.Equal(t, "Primary description\nDeprecated: use another type\ntrailing note", desc)
+	require.True(t, deprecated)
+}
+
+func TestExtractStructDocFallsBackToGenDecl(t *testing.T) {
+	ts := &ast.TypeSpec{ // no doc on type spec
+		Name: ast.NewIdent("Example"),
+	}
+
+	gd := &ast.GenDecl{
+		Doc: commentGroup(
+			"// +build ignore",
+			"// nolint",
```

Comment:
```suggestion
			"//nolint",
```

no lint usually does not have a whitespace and `isNolintDirective` expects it without one